### ISSUE TITLE
Zoom state?

### ISSUE THREAD

**Issue by dmitrirussu** [almende/vis#3027]
Do you have in your library the zoom state? How I can get the current zoom value state?

**Comment by wimrijnders** [comment]
For which part of `vis.js` would this be?

**Comment by dmitrirussu** [comment]
I want to do zoom by Slider and not by Buttons, if I give to Jq slider max zoom value "315360000000000" and min "10" the action slide is binding to mutch time and the VIS library stop to bind the zoomIn. slideer arrive to maxim value but zoom is not on 100%

**Comment by wimrijnders** [comment]
Is this for `network`, `timeline`, `graph2d` or `graph3d`?

**Comment by dmitrirussu** [comment]
Timeline

Create a fully customizable, interactive timeline with items and ranges.

**Comment by wimrijnders** [comment]
OK, clear. I hope that one of the timeline specialists can answer, then.

**Comment by dmitrirussu** [comment]
ok, thanks

**Comment by bradh** [comment]
I'd think about it as a callback on the rangechanged event. Maybe a simple jsfiddle would help to expand your question to make it a bit easier to understand.

**Comment by jgorene** [comment]
According to @bradh... a simple jsfiddle is often very useful;)
So, I also tested zoom by slider a few weeks ago and I encountered the same "problem" for too long execution time for each zoom and that works wrong using a slider finally. if we really talk about the same thing?
If this can help, Take a look on [this](https://output.jsbin.com/vipoci) ... just a base for testing for now and any suggestions are welcome;)

**Comment by jgorene** [comment]
@dmitrirussu , Thank you for you "Up" ;)
Just for info... does this answer your purpose?

**Comment by dmitrirussu** [comment]
@jgorene, yes is near, but is not working properly. anyway zoom up to 100% is not working. zoom out is ok.

**Comment by jgorene** [comment]
> zoom up to 100%, 
If you mean zoom out to 100% for maximum zoom out possible, have to adjust zoomMax options accordingly on this part of code below 
```
    // Using slider to zoomIn or zoomOut
    document.getElementById('sliderZoom').addEventListener('input', function(e) {
        var value = this.value;
        if (value < 0) {
            var start = moment().year(moment().year() - 100000), // to adjust with options
```
You can test and play with the same [example](https://output.jsbin.com/kajocutifo)...
It's just a working idea, maybe someone here can give another track ;)

**Comment by dmitrirussu** [comment]
@jgorene, thanks you it help me.

**Comment by cerensimsek1984** [comment]
my options ;
var options = {
		zoomMin: 1000 * 60 * 60 * 10,
		zoomMax: 1000 * 60 * 60 * 10 * 1000,             
		};

but i want to see first 1000 * 60 * 60 * 10 * 31 
how can i set this ? 
Timeline.zoomIn(1000 * 60 * 60 * 10 * 31);  ?

**Comment by brichbe** [comment]
@cerensimsek1984 I think maybe you're confusing zooming with the timeline window.

The timeline always displays some window (duration) of time.  Zooming simply adjusts that window of time so it's larger (zoom out) or smaller (zoom in) by a percentage of the currently visible window.  So what you've posted above doesn't make sense because it implies you're trying to zoom to a duration of time.

Instead, have a look at the [Timeline method](http://visjs.org/docs/timeline/#Methods) `setWindow`.  You provide a `start` and an `end` to adjust the visible window.  If you're not already familiar with using moment.js, you'll want to learn how to use that to help you do operations like this, so for example you can set the `start` to today and the `end` to today+31 days.

**Comment by cerensimsek1984** [comment]
Yes thank you. i understand now. I searched and succeed.. 

i added https://cdnjs.cloudflare.com/ajax/libs/moment.js/2.19.2/moment.min.js 

and later  https://jsfiddle.net/hc83Lpfw/5/     now i must set to the month of the selected year.

**Comment by ZaidRehman** [comment]
@jgorene Hey your example is cool but when user changes zoom by scrolling it doesn't update the slider.
Is there a way we can detect the current zoom level of the timeline to update the slider?
